third-party/llvm/Makefile.share-system

```diff
@@ -18,11 +18,11 @@ endif
 
 # LLVM preprocessor flags (ie -Dbla and -Ibla) 
 ifndef LLVM_CONFIG_CXXFLAGS
-  export LLVM_CONFIG_CXXFLAGS=$(shell test -x $(LLVM_CONFIG) && $(LLVM_CONFIG) --cxxflags | sed 's/ -O[0-4s] / /' | sed 's/ -DNDEBUG / /' | sed 's/ -pedantic / /' | sed 's/ -W / /' | sed 's/ -Wall / /')
+  export LLVM_CONFIG_CXXFLAGS=$(shell test -x $(LLVM_CONFIG) && $(LLVM_CONFIG) --cxxflags | xargs -n 1 | awk -f $(THIRD_PARTY_DIR)/llvm/filter-llvm-config.awk | xargs)
```

Comment:
If we're writing a script anyway, why not write a .sh / .bash script that returns the right stuff when passed the llvm-config arguments to run?
Separately, I'm not sure if xargs would be a new dependency if we use it here.

Reply:
It seems to me that an sh or bash script would be much more complicated than the awk script.  I don't think it would be worth it.

A standard POSIX environment is required to include xargs, though we could mention it the same place we mention awk.

Reply:
That's fair - I didn't already know it was part of the standard POSIX environment.